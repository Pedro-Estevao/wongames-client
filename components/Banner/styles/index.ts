'use client';

import styled, { css } from 'styled-components';

type ImageProps = {
	src: string;
};

export const BannerContainer = styled.div``;

export const Image = styled.div<ImageProps>`
	${({ src }) => css`
		background-image: url(${src});
		background-position: center center;
		background-size: cover;
	`}
`;
