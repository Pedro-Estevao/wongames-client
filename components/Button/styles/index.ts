'use client';

import styled, { css } from 'styled-components';
import { ButtonProps } from '..';
import { darken } from 'polished';

type ContainerModifiersProps = {
	hasIcon: boolean;
} & Pick<ButtonProps, 'size' | 'fullWidth' | 'minimal'>;

const containerModifiers = {
	withIcon: () => css`
		display: flex;
		align-items: center;
		justify-content: center;

		svg {
			width: 1.5rem;

			& + span {
				margin-left: 0.8rem;
			}
		}
	`,
	minimal: () => css`
		background: none;
		color: #f231a5;

		&:hover {
			color: ${darken(0.1, '#F231A5')};
		}
	`
};

export const ButtonContainer = styled.button<ContainerModifiersProps>`
	${({ hasIcon, minimal }) => css`
		${!!hasIcon && containerModifiers.withIcon};
		${!!minimal && containerModifiers.minimal};
	`}
`;
