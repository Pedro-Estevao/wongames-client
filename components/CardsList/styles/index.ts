import styled from 'styled-components';

export const CardsListContainer = styled.div``;

export const Card = styled.div`
	display: flex;
	align-items: center;
	background: #eaeaea;
	color: #030517;
	padding: 1.3rem 1.6rem;

	&:not(:last-child) {
		margin-bottom: 1.6rem;
	}
`;

export const CardLabel = styled.span`
	margin-left: 0.8rem;
`;
