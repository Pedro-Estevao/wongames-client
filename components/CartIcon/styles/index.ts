'use client';

import { darken } from "polished";
import styled, { css } from 'styled-components';

export const CartIconContainer = styled.button`
  ${({ theme }) => css`
		position: relative;
		background-color: transparent;
		border: none;
		cursor: pointer;
		color: ${theme.colors.white};
		display: flex;
		align-items: center;
		justify-content: center;
		width: ${theme.spacings.small};
		height: ${theme.spacings.small};
		transition: color ${theme.transition.default};

		&:hover {
			color: ${darken(0.3, theme.colors.white)};
		}
	`}
`;

export const Badge = styled.span`
	${({ theme }) => css`
		position: absolute;
		top: -0.4rem;
		right: -0.4rem;
		display: flex;
		align-items: center;
		justify-content: center;
		background-color: ${theme.colors.secondary};
		color: ${theme.colors.white};
		font-size: 1rem;
		border-radius: 50%;
		width: ${theme.spacings.xsmall};
		height: ${theme.spacings.xsmall};
	`}
`;
