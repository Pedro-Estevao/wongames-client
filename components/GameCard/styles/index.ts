'use client';

import styled, { css } from 'styled-components';

type PriceProps = {
	isPromotional?: boolean;
};

const priceModifiers = {
	default: () => css`
		color: #fafafa;
		padding: 0 0.8rem;
		background-color: #3cd3c1;
		border-radius: 0.4rem;
		margin-right: calc(0.8rem / 2);
	`,

	promotional: () => css`
		color: #8f8f8f;
		text-decoration: line-through;
		margin-right: 1.6rem;
	`
};

export const GameCardContainer = styled.article``;

export const ImageBox = styled.div`
	background-color: #f6f7f8;
	background-image: linear-gradient(
		to right,
		#f6f7f8 0%,
		#edeef1 20%,
		#f6f7f8 40%,
		#f6f7f8 100%
	);
	background-size: 80rem 14rem;
	animation: placeholderShimmer 1s linear infinite forwards;

	img {
		width: 100%;
		height: 100%;
		object-fit: cover;
	}

	@keyframes placeholderShimmer {
		0% {
			background-position: -40rem 0;
		}
		100% {
			background-position: 40rem 0;
		}
	}
`;

export const Content = styled.div``;

export const Info = styled.div`
	max-width: calc(100% - 2.5rem);
`;

export const Title = styled.h3`
	overflow: hidden;
	text-overflow: ellipsis;
	white-space: nowrap;
`;

export const Developer = styled.h4`
	word-wrap: break-word;
	overflow: hidden;
`;

export const FavButton = styled.div``;

export const BuyBox = styled.div``;

export const Price = styled.div<PriceProps>`
	${({ isPromotional }) => css`
		display: inline-flex;
		font-weight: 600;
		height: 3rem;
		align-items: center;

		${!isPromotional && priceModifiers.default};
		${isPromotional && priceModifiers.promotional};
	`}
`;
