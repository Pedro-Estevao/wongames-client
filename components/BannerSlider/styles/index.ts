'use client';

import styled, { css } from 'styled-components';
import * as BannerStyles from '@/components/Banner/styles';

export const BannerSliderContainer = styled.section`
	${() => css`
		.slick-dots {
			list-style: none;
			display: flex !important;
			align-items: center;
			justify-content: center;
			margin-top: 1.4rem;

			li {
				background: #fafafa;
				width: 1.2rem;
				height: 1.2rem;
				border-radius: 100%;
				display: flex;
				align-items: center;
				justify-content: center;
				margin: 0 0.8rem;
				cursor: pointer;
				&.slick-active {
					background: #f231a5;
				}
			}
			button {
				opacity: 0;
				width: 1.2rem;
				height: 1.2rem;
				cursor: pointer;
			}
		}

		@media (min-width: 1200px) {
			${BannerStyles.BannerContainer} {
				max-width: 104rem;
				margin: 0 auto;
			}

			.slick-dots {
				position: absolute;
				right: 0;
				flex-direction: column;
				height: 100%;
				top: 0;
				margin: 0;

				li {
					margin: 0.8rem 0;
				}
			}
		}
	`}
`;
