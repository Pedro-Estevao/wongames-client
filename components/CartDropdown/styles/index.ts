'use client';

import styled from 'styled-components';
import { CartListContainer as CartListStyle } from '@/components/CartList/styles';

export const CartDropdownContainer = styled.main`
	${CartListStyle} {
		width: 56rem;
	}
`;
