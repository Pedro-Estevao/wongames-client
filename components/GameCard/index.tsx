import React from 'react';
import Image from 'next/image';
import * as S from './styles';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import Button from '../Button';
import {
	faCartShopping,
	faHeart as heartSolid
} from '@fortawesome/free-solid-svg-icons';
import { faHeart as heartRegular } from '@fortawesome/free-regular-svg-icons';
import Ribbon, { RibbonColorsProps, RibbonSizeProps } from '../Ribbon';
import { formatPrice } from '@/utils/formats';

export type GameCardProps = {
	slug: string;
	title: string;
	developer: string;
	img: string;
	price: number | bigint;
	promotionalPrice?: number | bigint;
	favorite?: boolean;
	ribbon?: React.ReactNode;
	ribbonColor?: RibbonColorsProps;
	ribbonSize?: RibbonSizeProps;
	onFav?: () => void;
};

const GameCard = ({
	slug,
	title,
	developer,
	img,
	price,
	promotionalPrice,
	favorite = false,
	ribbon,
	ribbonColor = 'primary',
	ribbonSize = 'small',
	onFav
}: GameCardProps) => {
	return (
		<>
			<S.GameCardContainer
				className="relative flex flex-col w-full h-full rounded-[6px] overflow-hidden"
				href={`/game/${slug}`}
			>
				{!!ribbon && (
					<Ribbon color={ribbonColor} size={ribbonSize}>
						{ribbon}
					</Ribbon>
				)}
				<S.ImageBox className="h-[14rem] min-h-[14rem] overflow-hidden rounded-t-[6px] rounded-b-[0px] w-full">
					<Image src={img} width={300} height={140} alt={title} priority />
				</S.ImageBox>
				<S.Content className="relative flex flex-col justify-between h-full p-xsmall bg-white">
					<S.Info>
						<S.Title className="text-medium leading-medium font-bold text-black">
							{title}
						</S.Title>
						<S.Developer className="text-small font-bold text-gray">
							{developer}
						</S.Developer>
					</S.Info>
					<S.FavButton
						onClick={(e) => {
							e.preventDefault();
							onFav;
						}}
						className="absolute text-primary top-[1.6rem] cursor-pointer right-[1.2rem]"
						role="button"
					>
						{favorite ? (
							<FontAwesomeIcon
								icon={heartSolid}
								className="w-[2.5rem]"
								aria-label="Remove from Wishlist"
							/>
						) : (
							<FontAwesomeIcon
								icon={heartRegular}
								className="w-[2.5rem]"
								aria-label="Add to Wishlist"
							/>
						)}
					</S.FavButton>
					<S.BuyBox className="flex items-center justify-end mt-xxsmall">
						{!!promotionalPrice && (
							<S.Price isPromotional>{formatPrice(price)}</S.Price>
						)}
						<S.Price>
							{price === 0 ? 'FREE' : formatPrice(promotionalPrice || price)}
						</S.Price>
						<Button
							icon={<FontAwesomeIcon icon={faCartShopping} />}
							size="small"
						/>
					</S.BuyBox>
				</S.Content>
			</S.GameCardContainer>
		</>
	);
};

export default GameCard;
