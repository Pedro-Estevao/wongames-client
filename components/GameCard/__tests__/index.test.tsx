import { render, screen } from '@/utils/testsHelper';
import { expect } from '@jest/globals';
import GameCard, { GameCardProps } from "@/components/GameCard";
import { theme } from "@/public/styles/theme";

const props: GameCardProps = {
    id: '1',
    slug: 'population-zero',
    title: 'Population Zero',
    developer: 'Rockstar Games',
    img: 'https://images.unsplash.com/photo-1674461315270-d6b04eaff631',
    price: 235
};

describe('<GameCard />', () => {
    it('should render correctly', () => {
        const { container } = render(<GameCard {...props} />);

        expect(
            screen.getByRole('heading', { name: props.title })
        ).toBeInTheDocument();

        expect(
            screen.getByRole('heading', { name: props.developer })
        ).toBeInTheDocument();

        const image = screen.getByRole('img', { name: props.title });

        expect(image).toHaveAttribute('src');
        expect(image).toHaveAttribute('src', expect.stringContaining(encodeURIComponent(props.img)));

        expect(screen.getByRole('link')).toHaveAttribute('href', `/game/${props.slug}`);
        expect(screen.getByText(props.title)).toBeInTheDocument();

        expect(screen.getByLabelText(/add to wishlist/i)).toBeInTheDocument();

        expect(container.firstChild).toMatchSnapshot();
    });

    it('should render price in label', () => {
        render(<GameCard {...props} />);

        const price = screen.getByText('$235.00');

        expect(price).not.toHaveStyle({ textDecoration: 'line-through' });
        expect(price).toHaveStyle({ backgroundColor: theme.colors.secondary });
    })

    it('should render a line-through in price when promotional', () => {
        render(<GameCard {...props} promotionalPrice={15} />);

        expect(screen.getByText('$235.00')).toHaveStyle({
            textDecoration: 'line-through'
        });

        expect(screen.getByText('$15.00')).not.toHaveStyle({
            textDecoration: 'line-through'
        });
    });

    it('should render Ribbon', () => {
        render(
            <GameCard
                {...props}
                ribbon="My Ribbon"
                ribbonColor="secondary"
                ribbonSize="small"
            />
        );
        const ribbon = screen.getByText(/my ribbon/i);

        expect(ribbon).toHaveStyle({ backgroundColor: '#3CD3C1' });
        expect(ribbon).toHaveStyle({ height: '2.6rem', fontSize: '1.2rem' });
        expect(ribbon).toBeInTheDocument();
    });
});