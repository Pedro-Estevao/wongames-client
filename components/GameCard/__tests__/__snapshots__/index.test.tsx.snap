// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameCard /> should render correctly 1`] = `
.c0 {
  position: relative;
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100%;
  background-color: #FAFAFA;
  border-radius: 6px;
  text-decoration: none;
}

.c1 {
  position: relative;
  height: 14rem;
  min-height: 14rem;
  width: 100%;
  background-color: #f6f7f8;
  background-image: linear-gradient( to right,#f6f7f8 0%,#edeef1 20%,#f6f7f8 40%,#f6f7f8 100% );
  background-size: 80rem 14rem;
  overflow: hidden;
  animation: placeholderShimmer 1s linear infinite forwards;
}

.c1 img {
  object-fit: cover;
}

.c2 {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  position: relative;
  height: 100%;
  margin: 1.6rem;
}

.c3 {
  max-width: calc(100% - 2.5rem);
}

.c4 {
  font-size: 1.6rem;
  line-height: 1.6rem;
  font-weight: 600;
  color: #030517;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c5 {
  font-size: 1.4rem;
  font-weight: 600;
  color: #8F8F8F;
  word-wrap: break-word;
  overflow: hidden;
}

.c6 {
  position: absolute;
  right: -1rem;
  top: -0.5rem;
  color: #F231A5;
  cursor: pointer;
}

.c6 svg {
  width: 2.5rem;
}

.c8 {
  display: flex;
  align-items: center;
  justify-content: flex-end;
  margin-top: 0.8rem;
}

.c9 {
  display: inline-flex;
  font-weight: 600;
  height: 3rem;
  align-items: center;
  color: #FAFAFA;
  padding: 0 0.8rem;
  background-color: #3CD3C1;
  border-radius: 0.4rem;
  margin-right: calc(0.8rem / 2);
}

.c7 {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  color: #FAFAFA;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  border: 0;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.8rem;
  text-decoration: none;
  height: 3rem;
  font-size: 1.2rem;
  background: none;
  color: #F231A5;
}

.c7:focus {
  box-shadow: 0 0 0 3px #3CD3C1;
}

.c7:hover {
  background: none;
}

.c7 svg {
  width: 1.5rem;
}

.c7 svg+span {
  margin-left: 0.8rem;
}

.c7:hover {
  color: #e20e8d;
}

.c10 {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  color: #FAFAFA;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  border: 0;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.8rem;
  text-decoration: none;
  height: 3rem;
  font-size: 1.2rem;
}

.c10:focus {
  box-shadow: 0 0 0 3px #3CD3C1;
}

.c10:hover {
  background: linear-gradient(180deg, #e35565 0%, #d958a6 50%);
}

.c10 svg {
  width: 1.5rem;
}

.c10 svg+span {
  margin-left: 0.8rem;
}

<a
  class="c0"
  data-cy="game-card"
  href="/game/population-zero"
>
  <div
    class="c1 rounded-t-[6px] rounded-b-[0px]"
  >
    <img
      alt="Population Zero"
      data-nimg="fill"
      decoding="async"
      fetchpriority="high"
      sizes="100vw"
      src="/_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1674461315270-d6b04eaff631&w=3840&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1674461315270-d6b04eaff631&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1674461315270-d6b04eaff631&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1674461315270-d6b04eaff631&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1674461315270-d6b04eaff631&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1674461315270-d6b04eaff631&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1674461315270-d6b04eaff631&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1674461315270-d6b04eaff631&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1674461315270-d6b04eaff631&w=3840&q=75 3840w"
      style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
    />
  </div>
  <div
    class="c2"
  >
    <div
      class="c3"
    >
      <h3
        class="c4"
      >
        Population Zero
      </h3>
      <h4
        class="c5"
      >
        Rockstar Games
      </h4>
    </div>
    <div
      class="c6"
    >
      <button
        aria-label="Add to Wishlist"
        class="c7"
        style="filter: none;"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-heart "
          data-icon="heart"
          data-prefix="far"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M225.8 468.2l-2.5-2.3L48.1 303.2C17.4 274.7 0 234.7 0 192.8l0-3.3c0-70.4 50-130.8 119.2-144C158.6 37.9 198.9 47 231 69.6c9 6.4 17.4 13.8 25 22.3c4.2-4.8 8.7-9.2 13.5-13.3c3.7-3.2 7.5-6.2 11.5-9c0 0 0 0 0 0C313.1 47 353.4 37.9 392.8 45.4C462 58.6 512 119.1 512 189.5l0 3.3c0 41.9-17.4 81.9-48.1 110.4L288.7 465.9l-2.5 2.3c-8.2 7.6-19 11.9-30.2 11.9s-22-4.2-30.2-11.9zM239.1 145c-.4-.3-.7-.7-1-1.1l-17.8-20-.1-.1s0 0 0 0c-23.1-25.9-58-37.7-92-31.2C81.6 101.5 48 142.1 48 189.5l0 3.3c0 28.5 11.9 55.8 32.8 75.2L256 430.7 431.2 268c20.9-19.4 32.8-46.7 32.8-75.2l0-3.3c0-47.3-33.6-88-80.1-96.9c-34-6.5-69 5.4-92 31.2c0 0 0 0-.1 .1s0 0-.1 .1l-17.8 20c-.3 .4-.7 .7-1 1.1c-4.5 4.5-10.6 7-16.9 7s-12.4-2.5-16.9-7z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
    <div
      class="c8"
    >
      <div
        class="c9"
      >
        $235.00
      </div>
      <button
        aria-label="Add to cart"
        class="c10"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-cart-plus "
          data-icon="cart-plus"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 24C0 10.7 10.7 0 24 0L69.5 0c22 0 41.5 12.8 50.6 32l411 0c26.3 0 45.5 25 38.6 50.4l-41 152.3c-8.5 31.4-37 53.3-69.5 53.3l-288.5 0 5.4 28.5c2.2 11.3 12.1 19.5 23.6 19.5L488 336c13.3 0 24 10.7 24 24s-10.7 24-24 24l-288.3 0c-34.6 0-64.3-24.6-70.7-58.5L77.4 54.5c-.7-3.8-4-6.5-7.9-6.5L24 48C10.7 48 0 37.3 0 24zM128 464a48 48 0 1 1 96 0 48 48 0 1 1 -96 0zm336-48a48 48 0 1 1 0 96 48 48 0 1 1 0-96zM252 160c0 11 9 20 20 20l44 0 0 44c0 11 9 20 20 20s20-9 20-20l0-44 44 0c11 0 20-9 20-20s-9-20-20-20l-44 0 0-44c0-11-9-20-20-20s-20 9-20 20l0 44-44 0c-11 0-20 9-20 20z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
  </div>
</a>
`;
